pkg/graphql/request.go

```diff
@@ -179,6 +227,44 @@ func (r *Request) IsIntrospectionQuery() (result bool, err error) {
 	return true, nil
 }
 
+// IsIntrospectionQueryStrict returns true if the client tries to query __schema or __type fields in any way.
+// IsIntrospectionQuery returns false if schema/type introspection query contains additional non-introspection fields.
+// This breaks the granular access schema of Tyk Gateway.
+func (r *Request) IsIntrospectionQueryStrict() (result bool, err error) {
+	selectionSets, err := r.scanFragmentDefinitionsFindSelectionSets()
+	if err != nil {
+		return
+	}
+	selectionSet, err := r.scanOperationDefinitionsFindSelectionSet()
```

Comment:
You can redeclare err?

Reply:
And selectionSets?

Reply:
Yes, we can redeclare variables. It's safe because we already check the error before redeclaring. I found this in the Golang spec. 

> Unlike regular variable declarations, a short variable declaration may redeclare variables provided they were originally declared earlier in the same block (or the parameter lists if the block is the function body) with the same type, and at least one of the non-blank variables is new. […] Redeclaration does not introduce a new variable; it just assigns a new value to the original.

https://go.dev/ref/spec#Short_variable_declarations